lib/unleash/constraint.rb

```diff
@@ -107,7 +107,7 @@ def matches_constraint?(context)
       context_value = context.get_by_name(self.context_name)
 
       v.map!(&:upcase) if self.case_insensitive
-      context_value.upcase! if self.case_insensitive
+      context_value = context_value.upcase if self.case_insensitive
 
       OPERATORS[self.operator].call(context_value, v)
```

Comment:
would have been a prettier solution:

```suggestion
OPERATORS[self.operator].call(context_value.then{|c| self.case_insensitive ? c.upcase : c} , v.then{|c| self.case_insensitive ? c.upcase : c})
```

Reply:
note that using `!` methods in general is to be avoided, and specially on user provided data is dangerous, as you shouldn't be altering data provided / referenced by the user. (if he is to use again later on, it will be changed by this method in the SDK).

Reply:
> the same issue will reappear if the values in v are frozen.
Unless you're doing something super weird, the values in v should only come from internal code. If that starts to fail then we have bigger problems